presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -574,7 +578,7 @@ protected ListObjectsV2Result computeNext(ListObjectsV2Result previous)
         // user metadata, and in this case it doesn't matter.
         return objects.stream()
                 .filter(object -> !object.getKey().endsWith(PATH_SEPARATOR))
-                .filter(object -> !skipGlacierObjects || !isGlacierObject(object))
+                .filter(object -> !skipGlacierObjects || !GLACIER_STORAGE_CLASSES.contains(object.getStorageClass()))
```

Comment:
I think having the `isGlacierObject` method makes this easier to read. Up to you.

Reply:
Thanks for the review !! Retained that method